syntax = "proto3";

package poker;

option cc_generic_services = true;
option py_generic_services = true;

import "poker_messages.proto";

service PokerService {
  rpc IsReady(Empty) returns (Empty);
  rpc GetPlayers(Empty) returns (stream PlayerData);
  rpc GetDealer(stream PlayerData) returns (PlayerData);
  rpc GetPlayerAction (PlayerData) returns (ActionData);
  rpc GetPlayerCards (PlayerData) returns (CardsData);
  rpc GetBoardCards (BoardCardsRequest) returns (CardsData);
  rpc GetBlinds(PlayerData) returns (Amount);
  rpc PerformAction(ActionData) returns (Empty);
}